conan/tools/cmake/toolchain/blocks.py

```diff
@@ -843,16 +848,22 @@ def context(self):
         generator = self._toolchain.generator
         generator_platform = self._get_generator_platform(generator)
         toolset = self._get_toolset(generator)
-
-        compiler, compiler_cpp, compiler_rc = self._get_compiler(generator)
+        compilers_by_conf = self._conanfile.conf.get("tools.cmake.cmaketoolchain:compilers",
+                                                     default=[], check_type=list)
+        # FIXME: do we want to keep this legacy part?
+        compiler, compiler_cpp, compiler_rc = None, None, None
+        if not compilers_by_conf:
+            # Configuration has preference for setting compilers
+            compiler, compiler_cpp, compiler_rc = self._get_windows_compilers(generator)
```

Comment:
Should I keep this if the new approach is good?

Reply:
why is it named `_get_windows_compilers`? shouldn't it be platform-agnostic here?

Reply:
Sure, but the implementation looks like this:

```python
    def _get_windows_compilers(self, generator):
        compiler = self._conanfile.settings.get_safe("compiler")
        os_ = self._conanfile.settings.get_safe("os")

        compiler_c = compiler_cpp = compiler_rc = None
        # TODO: Check if really necessary now that conanvcvars is used
        if "Ninja" in str(generator) and is_msvc(self._conanfile):
            compiler_c = compiler_cpp = "cl"
        elif os_ == "Windows" and compiler == "clang" and "Visual" not in generator:
            # definition of compiler only if not using the VS Clang toolset
            compiler_rc = "clang"
            compiler_c = "clang"
            compiler_cpp = "clang++"

        return compiler_c, compiler_cpp, compiler_rc
```
So, I thought it could be a better name. I can keep the previous name for sure.

Reply:
well, there is FIXME above, maybe drop it, as it's legacy part?
otherwise, I think we previously agreed to use `blocks` approach, so it might have sense to isolate this kind of code into blocks like `MSVCNinja` and `ClangWindows` (if it's really needed).
we know, the name of compiler might be different, such as `clang-13.exe` or `i686-w64-mingw32-clang++.exe`

Reply:
I think putting this in a dedicated ``CompilerBlock`` or something like that (just 1 block for the definition of this, not one block per use case), makes sense.

Reply:
It sounds good to me!